src/main/java/org/jabref/logic/citationkeypattern/CitationKeyGenerator.java

```diff
@@ -213,7 +213,7 @@ private String createCitationKeyFromPattern(BibEntry entry) {
             // ":lower":
             if (fieldParts.size() > 1) {
                 // apply modifiers:
-                expandedPattern = applyModifiers(expandedPattern, fieldParts, 1);
+                expandedPattern = applyModifiers(expandedPattern, fieldParts, 1, expandBracketContent(keywordDelimiter, entry, database));
```

Comment:
You are using `BracketedPattern.expandBracketContent` instead of `CitationKeyGenerator.expandBracketContent(BibEntry)` here.